spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/discovery/KubernetesDiscoveryProperties.java

```diff
@@ -51,14 +52,15 @@ public record KubernetesDiscoveryProperties(
 		@DefaultValue({"443", "8443"}) Set<Integer> knownSecurePorts,
 		@DefaultValue Map<String, String> serviceLabels, String primaryPortName,
 		@DefaultValue Metadata metadata,
-		@DefaultValue("" + DEFAULT_ORDER) int order) {
+		@DefaultValue("" + DEFAULT_ORDER) int order,
+		@DefaultValue("true") boolean useEndpointSlices) {
```

Comment:
1. we enable using `EndpointSlices` by default. Users can opt-out of this one, and use Endpoints only - for example if their cluster is on a version where `EndpointSlices` are not supported. 

 - Let's assume you start the app with a version of k8s where EndpointSlices are _not_ supported : we will default to using `Endpoints` in such case (even if `useEndpointSlices=true`).
 - If you start the app and EndpointSlices are supported, we will use it.

So the question might come up of why then do we need such a flag, if we will default anyway? Well, my thought process is this:

- for EndpointSlices to work, you need proper `Role` settings and that might not always be an option. So I thought that if changing the Role si not an option, you can disable EndpointSlices, even if the cluster supports them.